nmcp/src/main/kotlin/nmcp/transport/transport.kt

```diff
@@ -125,7 +125,18 @@ internal class HttpTransport(
                 client.newCall(it).execute()
             }.use { response ->
                 check(response.isSuccessful) {
-                    "Nmcp: cannot PUT '$url' (statusCode=${response.code}):\n${response.body!!.string()}"
+                    buildString {
+                        appendLine("Nmcp: cannot PUT '$url' (statusCode=${response.code}).")
+                        appendLine("Response body: ${response.body!!.string()}")
+                        appendLine("Things to double check:")
+                        // I have seen 401 for this
+                        appendLine(" - Are your credentials correct?")
+                        appendLine(" - Did you enable the snapshots on your namespace at https://central.sonatype.com/publishing/namespaces?")
+                        // I have seen 400 for this
+                        appendLine(" - Is your version ending with `-SNAPSHOT`?")
```

Comment:
The plugin could check the version rather than guess

Reply:
Right, I have added this to https://github.com/GradleUp/nmcp/issues/82